command/ca/provisionerbeta/add.go

```diff
@@ -0,0 +1,462 @@
+			},
+			cli.StringFlag{
+				Name:  "ssh-template",
+				Usage: `The x509 certificate template <file>, a JSON representation of the certificate to create.`,
+			},
+
+			// JWK provisioner flags
+			cli.BoolFlag{
+				Name:  "create",
+				Usage: `Create the JWK key pair for the provisioner.`,
+			},
+			cli.StringFlag{
+				Name:  "private-key",
+				Usage: `The <file> containing the JWK private key.`,
+			},
+
+			// OIDC provisioner flags
+			cli.StringFlag{
+				Name:  "client-id",
+				Usage: `The <id> used to validate the audience in an OpenID Connect token.`,
+			},
+			cli.StringFlag{
+				Name:  "client-secret",
+				Usage: `The <secret> used to obtain the OpenID Connect tokens.`,
+			},
+			cli.StringFlag{
+				Name:  "listen-address",
+				Usage: `The callback <address> used in the OpenID Connect flow (e.g. \":10000\")`,
+			},
+			cli.StringFlag{
+				Name:  "configuration-endpoint",
+				Usage: `OpenID Connect configuration <url>.`,
+			},
+			cli.StringSliceFlag{
+				Name: "admin",
+				Usage: `The <email> of an admin user in an OpenID Connect provisioner, this user
+will not have restrictions in the certificates to sign. Use the
+'--admin' flag multiple times to configure multiple administrators.`,
+			},
+
+			// X5C provisioner flags
+			cli.StringFlag{
+				Name: "x5c-root",
+				Usage: `Root certificate (chain) <file> used to validate the signature on X5C
+provisioning tokens.`,
+			},
+			// K8sSA provisioner flags
+			cli.StringFlag{
+				Name: "pem-keys",
+				Usage: `Public key <file> for validating signatures on K8s Service Account Tokens.
+PEM formatted bundle (can have multiple PEM blocks in the same file) of public
+keys and x509 Certificates.`,
+			},
+			flags.X5cCert,
+			flags.X5cKey,
+			flags.PasswordFile,
+			flags.CaURL,
+			flags.Root,
+		},
+		Description: `**step ca provisioner add** adds a provisioner to the CA configuration.
+
+## POSITIONAL ARGUMENTS
+
+<name>
+: The name of the provisioner.
+
+<type>
+: The <type> of provisioner to create.
+
+	<type> is a case-insensitive string and must be one of:
+
+    **JWK**
+    : Uses an JWK key pair to sign provisioning tokens. (default)
+
+    **OIDC**
+    : Uses an OpenID Connect provider to sign provisioning tokens.
+
+    **AWS**
+    : Uses Amazon AWS instance identity documents.
+
+    **GCP**
+    : Use Google instance identity tokens.
+
+    **Azure**
+    : Uses Microsoft Azure identity tokens.
+
+    **ACME**
+    : Uses the ACME protocol to create certificates.
+
+    **X5C**
+    : Uses an X509 Certificate / private key pair to sign provisioning tokens.
+
+    **K8sSA**
+    : Uses Kubernetes Service Account tokens.
+
+    **SSHPOP**
+    : Uses an SSH Certificate / private key pair to sign provisioning tokens.
+
+## EXAMPLES
+
+Create a JWK provisioner:
+'''
+step ca provisioner add jane@doe.com --type JWK --ssh --public-key jwk.pub --private-key jwk.priv
```

Comment:
Not like this, and looking at this, I'm not sure if we want to add the --type back. Or change it to `step ca provisioned add JWK jane@doe.com ...`

Reply:
Beta.